Content.Server/Traits/Assorted/NarcolepsySystem.cs

```diff
@@ -42,22 +64,52 @@ public override void Update(float frameTime)
         var query = EntityQueryEnumerator<NarcolepsyComponent>();
         while (query.MoveNext(out var uid, out var narcolepsy))
         {
+            if (HasComp<SleepingComponent>(uid))
+                continue;
+
             narcolepsy.NextIncidentTime -= frameTime;
+            if (narcolepsy.NextIncidentTime <= narcolepsy.TimeBeforeWarning && narcolepsy.NextIncidentTime < narcolepsy.LastWarningRollTime - 1f)
+            {
+                // Roll for showing a popup. There should really be a class for doing this.
+                narcolepsy.LastWarningRollTime = narcolepsy.NextIncidentTime;
+                if (_random.Prob(narcolepsy.WarningChancePerSecond))
+                {
+                    ShowRandomPopup(uid, narcolepsy.WarningLocaleBase, narcolepsy.WakeupLocaleCount);
+                    narcolepsy.LastWarningRollTime = 0f; // Do not show any more popups for the upcoming incident
+                }
+            }
 
             if (narcolepsy.NextIncidentTime >= 0)
                 continue;
 
-            // Set the new time.
-            narcolepsy.NextIncidentTime +=
-                _random.NextFloat(narcolepsy.TimeBetweenIncidents.X, narcolepsy.TimeBetweenIncidents.Y);
+            StartIncident((uid, narcolepsy));
+        }
+    }
 
-            var duration = _random.NextFloat(narcolepsy.DurationOfIncident.X, narcolepsy.DurationOfIncident.Y);
+    public void StartIncident(Entity<NarcolepsyComponent> ent)
+    {
+        var duration = _random.NextFloat(ent.Comp.DurationOfIncident.X, ent.Comp.DurationOfIncident.Y);
+        PrepareNextIncident(ent, duration);
 
-            // Make sure the sleep time doesn't cut into the time to next incident.
-            narcolepsy.NextIncidentTime += duration;
+        _statusEffects.TryAddStatusEffect<ForcedSleepingComponent>(ent, StatusEffectKey, TimeSpan.FromSeconds(duration), false);
+    }
```

Comment:
`TryAddStatusEffect` can fail so I think I'd want this to be a bool with error logging

Reply:
check inserted (probably) but even your addition thing doesnt have tryaddstatus reporting an error log so i am fairly lost